### PR TITLE
CP-14879: hide Hyperliquid networks (HyperEVM/HyperCore) behind hyperliquid-support feature flag

### PR DESCRIPTION
**Pull request by B0Y3R-AVA**
## Description

**Ticket: [CP-14879](https://ava-labs.atlassian.net/browse/CP-14879)**

The backend started serving HyperEVM (chainId 999) — and potentially HyperCore (9999) — in `PROXY_URL/networks` as part of the extension team's Hyperliquid launch (CP-14533 / CP-14848, `display_in_core` flipped in core-token-aggregator). Mobile has no Hyperliquid support yet (CP-14595), but we render everything that endpoint returns, so HyperEVM appeared under "Avalanche L1s" in Manage Networks — and users with prior HyperEVM activity got it auto-enabled with balances via Glacier `listAddressChains`. Extension/web protect themselves with a client-side PostHog gate; mobile had none.

This PR:

- Adds a new PostHog feature gate `hyperliquid-support` (`FeatureGates.HYPERLIQUID_SUPPORT`, default **OFF**), following the `solana-support` pattern
- Strips Hyperliquid networks (chain ids 999/9999, plus chain-name match as defense) at the **read layer**: react-query `select` in `useGetNetworks` plus the same shared `filterOutHyperliquidNetworks` predicate in `getNetworksFromCache` (which backs the redux selectors). The cached `/networks` response stays unfiltered and the query key stays `[NETWORKS, includeSolana]`, so flipping the flag re-derives instantly from cache — no refetch, no cache fragmentation
- New `utils/network/isHyperliquidNetwork.ts` helpers (+ tests)
- Manage Networks: Hyperliquid chains are excluded from the "Avalanche L1s" bucket and render in their own "Hyperliquid" section — which only appears once the flag is enabled

No new dependencies. Not breaking: users who already toggled HyperEVM on keep an inert `enabledChainIds` entry that resolves to `undefined` and is filtered out safely.

## Screenshots/Videos
FF OFF
Screenshot 2026-07-27 at 5 04 08 PM

FF ON
Screenshot 2026-07-27 at 4 59 26 PM

## Testing

**Dev Testing (if applicable)**

IOS: 9416
Android: 9417

- Happy path: launch the app with default flags → Settings → Networks → HyperEVM/HyperCore must NOT appear anywhere (search included); portfolio must not show HyperEVM balances even for accounts with HyperEVM activity
- Flag on: enable `hyperliquid-support` (+ `everything`) in PostHog → Networks screen shows a "Hyperliquid" section containing HyperEVM, NOT listed under "Avalanche L1s"
- Edge case: account that had HyperEVM toggled on before this fix → no crash, network silently hidden while flag is off
- Unit tests: `NetworkService.test.ts` (Hyperliquid include/exclude), `isHyperliquidNetwork.test.ts`, `store/network/slice.test.ts` — all passing (172 tests across touched suites)

**QA Testing (if applicable)**

- Verify HyperEVM is gone from Manage Networks, network selector, send flow, and portfolio with production flags
- Verify a wallet with HyperEVM transaction history does not auto-enable or display the network
- Acceptance: HyperEVM/HyperCore are invisible in Core Mobile until `hyperliquid-support` is enabled in PostHog

## Checklist

Please check all that apply (if applicable)

- [x] I have performed a self-review of my code
- [x] I have verified the code works
- [ ] I have included screenshots / videos of android and ios
- [x] I have added testing steps
- [x] I have added/updated necessary unit tests
- [ ] I have updated the documentation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[CP-14879]: https://ava-labs.atlassian.net/browse/CP-14879?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core-mobile/app/hooks/networks/utils/getNetworksFromCache.ts [hunk 1]
```diff
@@ -3,11 +3,13 @@ import { queryClient } from 'contexts/ReactQueryProvider'
...
+  includeHyperliquid = false
 }: {
   includeSolana: boolean
+  includeHyperliquid: boolean
 }): Networks | undefined => {
   return queryClient.getQueryCache().find({
-    queryKey: [ReactQueryKeys.NETWORKS, includeSolana]
+    queryKey: [ReactQueryKeys.NETWORKS, includeSolana, includeHyperliquid]
   })?.state.data as Networks | undefined
 }
```
[the remark this answers is not in the document]
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/hooks/networks/utils/getNetworksFromCache.ts:23]: Done in d1c2acf — switched to `queryClient.getQueryData` with the exact key and dropped the dead defaults (both call sites pass explicitly, so params stay required).

File: packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  if (!network) {
+    return false
+  }
+
+  return (
+    isHyperliquidChainId(network.chainId) ||
+    network.chainName === HYPEREVM_CHAIN_NAME ||
+    network.chainName === HYPERCORE_CHAIN_NAME
+  )
+}
```
[the remark this answers is not in the document]
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts:27]: Done in d1c2acf — chain-name fallback now trims and compares case-insensitively (still exact token match, so no false-positive risk).

File: packages/core-mobile/app/store/network/slice.ts [hunk 3]
```diff
@@ -152,9 +155,11 @@ export const selectActiveNetwork = (state: RootState): Network => {
...
 export const selectAllNetworks = (state: RootState): Networks => {
   const isSolanaSupportBlocked = selectIsSolanaSupportBlocked(state)
+  const isHyperliquidSupportBlocked = selectIsHyperliquidSupportBlocked(state)
   const customNetworks = selectCustomNetworks(state)
   const rawNetworks = getNetworksFromCache({
-    includeSolana: !isSolanaSupportBlocked
+    includeSolana: !isSolanaSupportBlocked,
+    includeHyperliquid: !isHyperliquidSupportBlocked
   })
   return { ...rawNetworks, ...customNetworks }
```
[the remark this answers is not in the document]
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/store/network/slice.ts:164]: Object spread of `undefined` is safe — `{...undefined, ...customNetworks}` evaluates to `{...customNetworks}` per spec (CopyDataProperties ignores null/undefined). The TypeError you describe applies to `Object.keys(undefined)`/`Object.assign`, not spread syntax, and this spread pattern predates this PR unchanged. Verified: `node -e 'console.log({...undefined, a: 1})'` → `{ a: 1 }`.

File: packages/core-mobile/app/store/network/slice.test.ts [hunk 4]
```diff
@@ -32,7 +32,8 @@ jest.mock('hooks/networks/utils/getNetworksFromCache', () => ({
 }))
 
 jest.mock('store/posthog/slice', () => ({
-  selectIsSolanaSupportBlocked: jest.fn(() => false)
+  selectIsSolanaSupportBlocked: jest.fn(() => false),
+  selectIsHyperliquidSupportBlocked: jest.fn(() => true)
 }))
```
[the remark this answers is not in the document]
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/store/network/slice.test.ts:37]: The mock demonstrably applies through the barrel: `store/posthog` re-exports `./slice`, so `jest.mock('store/posthog/slice')` intercepts the underlying module. Empirical proof from this very PR — before the new selector was added to this mock, the suite failed with `selectIsHyperliquidSupportBlocked is not a function`, i.e. the sparse mock was being hit. Mocking the leaf module is also more precise than mocking the barrel, which would sparse-mock every other consumer of `store/posthog` in the test's dependency tree.

**Review by ruijia1in — COMMENTED** [review]
## Adversarial review (self-review pass)

Red-teamed the full diff plus both cache read paths (redux selectors + the react-query hook), the enabled-networks/balance derivation, and how `includeSolana` differs from the new filter.

**No critical/high issues** — flag defaults OFF, query keys line up across all four cache readers, and a stale enabled HyperEVM id resolves to `undefined` and drops out of `selectEnabledNetworks` with no crash.

### Main takeaway (design)
`includeHyperliquid` is modeled as a fetch/cache parameter, but unlike `includeSolana` (a real backend param: `/networks?includeSolana`) it's a **pure client-side filter** applied after the response. Putting it in the query key:
- triggers a full `/networks` + deBank refetch on flag flip even though the request is byte-for-byte identical, and
- fragments the cache into two entries that differ only by a post-filter, and
- forces the boolean to thread through ~6 files, growing combinatorially with every future gated network.

**Cleaner / more future-proof:** filter at the **read layer** instead. Leave `getNetworks`/`useGetNetworks`/`getNetworksFromCache`/`BalanceManager` and the query key keyed only on `includeSolana`, and apply one shared `excludeHyperliquid(networks)` predicate at the two spots that already read the PostHog flag — `useNetworks` and `selectNetworks`/`selectAllNetworks`. This still fixes the auto-enable/balances bug (verified: `selectEnabledNetworks` looks up `networks[chainId]`, so stripping the map drops the stale id), avoids the refetch + cache fragmentation, and makes the next gated network a one-line predicate instead of new plumbing. Inline notes below.

### Reviewed clean
- Security: no secrets/PII/logging, no injection surface, no new deps.
- Mutation-during-iteration: `Object.entries()` snapshots keys, so the in-place `delete` is safe.
- `getQueryData` swap (from `.find()`): more correct now that the key has 3 segments; all callers pass all 3.
- React memo deps in `ManageNetworksScreen` are correct; the Hyperliquid section is gated implicitly via an empty `networks` map when the flag is off.

File: packages/core-mobile/app/hooks/networks/useGetNetworks.ts [hunk 5]
```diff
@@ -4,15 +4,18 @@ import NetworkService from 'services/network/NetworkService'
...
-  includeSolana
+  includeSolana,
+  includeHyperliquid
 }: {
   includeSolana: boolean
+  includeHyperliquid: boolean
 }): UseQueryResult<Networks, Error> => {
   return useQuery({
-    queryKey: [ReactQueryKeys.NETWORKS, includeSolana],
+    queryKey: [ReactQueryKeys.NETWORKS, includeSolana, includeHyperliquid],
```
Comment (ruijia1in) [inline packages/core-mobile/app/hooks/networks/useGetNetworks.ts]: **[Medium — design]** `includeHyperliquid` is a client-side filter, not a backend param (only `includeSolana` changes the actual request, `/networks?includeSolana`). Including it in the query key fragments the cache and forces a needless full refetch (`/networks` + deBank) on flag flip. Consider keeping the key as `[NETWORKS, includeSolana]` and filtering Hyperliquid at the read layer (`useNetworks` + `selectNetworks`/`selectAllNetworks`) with a shared predicate instead — same bug fix, no cache fragmentation, and future gated networks become a one-line predicate rather than another threaded boolean.

File: packages/core-mobile/app/services/network/NetworkService.ts [hunk 6]
```diff
@@ -54,6 +60,20 @@ class NetworkService {
 
     delete networks[ChainId.AVALANCHE_LOCAL_ID]
 
+    if (!includeHyperliquid) {
+      // Hyperliquid networks (HyperEVM/HyperCore) are served by the backend for
+      // Core extension/web but are not supported on mobile yet — hide them
+      // unless the hyperliquid-support feature flag is enabled
+      Object.entries(networks).forEach(([chainId, network]) => {
+        if (
+          isHyperliquidChainId(Number(chainId)) ||
```
Comment (ruijia1in) [inline packages/core-mobile/app/services/network/NetworkService.ts]: **[Low — redundant]** `isHyperliquidChainId(Number(chainId))` is dead here: `isHyperliquidNetwork(network)` on the next line already re-checks `network.chainId`. One of the two is enough.

File: packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  if (!network) {
+    return false
+  }
+
+  const normalizedChainName = network.chainName?.trim().toLowerCase()
+
+  return (
+    isHyperliquidChainId(network.chainId) ||
+    normalizedChainName === HYPEREVM_CHAIN_NAME.toLowerCase() ||
+    normalizedChainName === HYPERCORE_CHAIN_NAME.toLowerCase()
```
Comment (ruijia1in) [inline packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts:25]: **[Medium — name match on user data]** The chain-*name* fallback does fuzzy identity matching. A custom network a user named "HyperEVM"/"HyperCore" would match, get bucketed into the Hyperliquid section, and be excluded from Avalanche L1s in `ManageNetworksScreen`. Consider restricting the name fallback to non-custom networks, or dropping it and treating chain id (999/9999) as the source of truth.

File: packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import { Network } from '@avalabs/core-chains-sdk'
+
+export const HYPEREVM_CHAIN_ID = 999
+// synthetic Core list id used by the backend for HyperCore — not an EVM chain id
+export const HYPERCORE_CHAIN_ID = 9999
```
Comment (ruijia1in) [inline packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts:6]: **[Low — verify the id]** The PR body says "potentially HyperCore (9999)". If the backend never serves 9999 this branch is dead code; if it serves HyperCore under a different synthetic id, the id check silently misses and we fall back to the name match. Worth confirming the actual id with the extension team before merge.

File: packages/core-mobile/app/hooks/networks/useGetNetworks.ts (hunk 5 above)
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/hooks/networks/useGetNetworks.ts]: Good call — done in 2debcaa. Query key is back to `[NETWORKS, includeSolana]` and the cache stays unfiltered; gating now happens at the two read choke points: react-query `select` in `useGetNetworks` (memoized on the flag, so a flip re-derives instantly with no refetch) and the same `filterOutHyperliquidNetworks` predicate in `getNetworksFromCache` for the redux selectors. Future gated networks are a one-line addition to the predicate.

File: packages/core-mobile/app/services/network/NetworkService.ts (hunk 6 above)
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/services/network/NetworkService.ts]: Done in 2debcaa — the fetch-time block is gone entirely with the read-layer refactor; the predicate now checks the network object only (the record key always equals `network.chainId` by construction, since `fetchNetworks` builds the map via `acc[network.chainId] = network`).

File: packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts (hunk 7 above)
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts:25]: Custom networks structurally can't be hidden by this: the filter only runs on the raw `/networks` cache (`useGetNetworks` select + `getNetworksFromCache`), and customs merge in *after* — `useNetworks` and `selectNetworks`/`selectAllNetworks` spread `customNetworks` over the filtered raw map. In `ManageNetworksScreen` both the L1s and Hyperliquid sections exclude customs before the name predicate runs, so a custom network named "HyperEVM" stays in Custom networks regardless of the flag. Kept the name fallback for backend data since HyperCore's synthetic id isn't confirmed (see your other comment) — this also matches the extension, which whitelists custom-added Hyperliquid networks even with their flag off.

File: packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts (hunk 8 above)
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/utils/network/isHyperliquidNetwork.ts:6]: 9999 is the extension's own constant for the synthetic Core-list entry (`HYPERCORE_CHAIN_ID = 9999` in core-extension `packages/common/src/utils/network/isHyperliquidNetwork.ts`, alongside caip `hlcore:mainnet`) — not something I invented. If BE serves HyperCore under a different id, the name fallback catches the current naming; if it never serves it through `/networks`, the branch is inert. Either way flag-off behavior is safe. I'll confirm the actual payload with the extension team (Bao) before we ever enable `hyperliquid-support` on mobile — tracked under CP-14595.

File: packages/core-mobile/app/new/features/accountSettings/screens/manageNetworks/ManageNetworksScreen.tsx [hunk 9]
```diff
@@ -115,7 +116,17 @@ export const ManageNetworksScreen = (): JSX.Element => {
           !alwaysEnabledChainIds.includes(network.chainId) &&
           !custom.some(item => item.chainId === network.chainId) &&
           !defaultEnabledL2ChainIds.includes(network.chainId) &&
-          !isXChain(network.chainId)
+          !isXChain(network.chainId) &&
+          !isHyperliquidNetwork(network)
+      )
+    )
+    // Hyperliquid networks are not Avalanche L1s — when the hyperliquid-support
+    // feature flag is off they never reach this screen at all
```
[the remark this answers is not in the document]
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/new/features/accountSettings/screens/manageNetworks/ManageNetworksScreen.tsx]: Done in 29745b9 — comment now says backend-provided Hyperliquid networks are stripped upstream while the flag is off, and user-added customs always render in the Custom networks section.

File: packages/core-mobile/app/new/features/accountSettings/screens/manageNetworks/ManageNetworksScreen.tsx [hunk 10]
```diff
@@ -115,7 +116,17 @@ export const ManageNetworksScreen = (): JSX.Element => {
           !alwaysEnabledChainIds.includes(network.chainId) &&
           !custom.some(item => item.chainId === network.chainId) &&
           !defaultEnabledL2ChainIds.includes(network.chainId) &&
-          !isXChain(network.chainId)
+          !isXChain(network.chainId) &&
+          !isHyperliquidNetwork(network)
+      )
+    )
```
[the remark this answers is not in the document]
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/new/features/accountSettings/screens/manageNetworks/ManageNetworksScreen.tsx:122]: Done in 29745b9 — the L1s section now uses the already-filtered `layer1Networks` directly (the double `filterNetworks` predated this PR, but agreed it was wasteful).

File: packages/core-mobile/app/store/posthog/types.ts [hunk 11]
```diff
@@ -63,7 +63,10 @@ export const DefaultFeatureFlagConfig = {
   [FeatureGates.DELEGATION_FEE_ENABLED]: false,
   // CP-13903 kill-switch. `false` = feature fully off: balance requests keep
   // dust, tx building keeps dust, Settings row hidden.
-  [FeatureGates.FILTER_SMALL_UTXOS]: false
+  [FeatureGates.FILTER_SMALL_UTXOS]: false,
+  // Hyperliquid networks kill-switch. `false` = HyperEVM/HyperCore are
+  // stripped from the network list entirely.
```
[the remark this answers is not in the document]
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/store/posthog/types.ts]: Done in 29745b9 — comment now states both `hyperliquid-support` AND `everything` must be enabled, pointing at `selectIsHyperliquidSupportBlocked`.

File: packages/core-mobile/app/hooks/networks/useGetNetworks.ts [hunk 12]
```diff
@@ -1,19 +1,33 @@
...
+  // hyperliquid-support flag re-derives instantly instead of refetching
+  const select = useCallback(
+    (networks: Networks): Networks =>
+      includeHyperliquid ? networks : filterOutHyperliquidNetworks(networks),
+    [includeHyperliquid]
+  )
+
   return useQuery({
     queryKey: [ReactQueryKeys.NETWORKS, includeSolana],
     queryFn: () =>
```
[the remark this answers is not in the document]
Comment (B0Y3R-AVA) [reply packages/core-mobile/app/hooks/networks/useGetNetworks.ts:26]: Good catch — the description predated the read-layer refactor (2debcaa). PR body updated to describe the implemented behavior: unfiltered cache, 2-element key, select-based gating, flag flips re-derive without refetch.

**Review by ruijia1in — APPROVED** [review]

**Review by eunjisong — APPROVED** [review]
